frontend/test/__support__/e2e/helpers/e2e-misc-helpers.js

```diff
@@ -39,3 +39,27 @@ export function runNativeQuery(xhrAlias = "dataset") {
   cy.wait("@" + xhrAlias);
   cy.icon("play").should("not.exist");
 }
+
+/**
+ * Intercepts a request and returns resolve function that allows
+ * the request to continue
+ *
+ * @param {string} method - Request method ("GET", "POST", etc)
+ * @param {string} path - Request URL to intercept
+ * @example
+ * const req = interceptPromise("GET", "/dashboard/1");
+ * // ... do something before request is allowed to go through ...
+ * req.resolve();
+ */
+export function interceptPromise(method, path) {
```

Comment:
Looks nice to me, I wonder if @nemanjaglumac knows about something similar we may already use.

Reply:
I looked for `cy.intercept` which is used all over the place but for deterministically waiting for requests to complete rather than intercepting them and controlling their flow. The only two helpers with `intercept` use it for waiting for a specific card query endpoint and modifying the response object (no flow control).

I also checked for the deprecated `cy.route` called but it was not used in any helpers I could find